### PR TITLE
Add link serialize

### PR DESCRIPTION
**Pull request by superstar54**
Currently, the AiiDA serializer manages to serialize a link (using `!!python/object` tag), but deserializing it leads to infinite recursion. The problem is that the `__init__` of the `NodeLinksManager` is not called by yaml, see this issue: https://github.com/yaml/pyyaml/issues/216.

This PR defines the yaml constructor and represent for the link.

### CODE REVIEW

File: tests/orm/utils/test_serialize.py [hunk 1]
```diff
@@ -179,3 +179,12 @@ def test_dataclass():
...
     deserialized = serialize.deserialize_unsafe(serialized)
     assert deserialized == obj
+
+
+def test_serialize_link():
+    """Test you can serialize and deserialize a link"""
+    from aiida.orm.utils.managers import NodeLinksManager
+    node = orm.Data().store()
+    link = NodeLinksManager(node=node, link_type=LinkType.CREATE, incoming=False)
+    deserialized = serialize.deserialize_unsafe(serialize.serialize(link))
```
Comment (sphuber) [inline tests/orm/utils/test_serialize.py:189]: I would assert `isinstance(deserialized, NodeLinksManager)` here and asser that the `node`, `link_type` and `incoming` attributes are as they were before

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @superstar54 . Changes look good in principle. I would just not call it "link" everywhere, because the `NodeLinksManager` is not a link really. So could you please update the names everywhere and change `link` explicitly to `node_links_manager`.

File: tests/orm/utils/test_serialize.py (hunk 1 above)
Comment (superstar54) [reply tests/orm/utils/test_serialize.py]: resolved in commit [d65362f](https://github.com/aiidateam/aiida-core/pull/6199/commits/d65362fcfc062458a08df51b4cb8541e25e2aa15)

**Comment by superstar54** [comment]
Hi @sphuber , I made the changes as you suggested. 
Regarding the failed test, should I create a `to_dict` method in the `NodeLinksManager`?

**Comment by superstar54** [comment]
I added `# pylint: disable`. This PR is ready for review.

**Review by sphuber — APPROVED** [review]
Thanks @superstar54 !

**Pull request merged**